### PR TITLE
Use pytest instead of nose3

### PR DESCRIPTION
**Pull request by jarrodmillman**

### CODE REVIEW

**Comment by matthew-brett** [comment]
I think this one's ready to merge now.  Do y'all agree?

**Comment by matthew-brett** [comment]
OK - I think this one's good to go - I've done some more refactoring of legacy nose stuff.   Will merge when tests pass.

**Comment by effigies** [comment]
This seems to lose the coverage reports entirely.

**Comment by matthew-brett** [comment]
> This seems to lose the coverage reports entirely.

Fixed now?

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -26,21 +26,20 @@ jobs:
       - name: Install
         run: |
           pip install -r dev-requirements.txt
-          pip install .
+          pip install -e .
```
[the remark this answers is not in the document]
Comment (matthew-brett) [reply .github/workflows/coverage.yml:29]: Backed off to not-editable.

File: nipy/fixes/__pycache__/setup.cpython-310.pyc
[the remark this answers is not in the document]
Comment (matthew-brett) [reply nipy/fixes/__pycache__/setup.cpython-310.pyc:1]: Good spot - I've rebased to remove these, and added `__pycache__` to the `.gitignore`.

**Review by effigies — APPROVED** [review]
LGTM

**Comment by effigies** [comment]
Well that `__config__` one was a surprise. I assume that will be purged with the meson build?

**Comment by matthew-brett** [comment]
I've skipped the `__config__.py` files in the doctests for now.  Merging - thanks for the review.   Now on to the Meson build ...

**Pull request merged**